src/main/java/no/fint/betaling/controller/FileController.java

```diff
@@ -0,0 +1,56 @@
+package no.fint.betaling.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import no.fint.betaling.exception.NoVISIDColumnException;
+import no.fint.betaling.exception.UnableToReadFileException;
+import no.fint.betaling.service.FileService;
+import no.fint.betaling.util.CustomerFileGroup;
+import org.apache.tika.Tika;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.FileInputStream;
+import java.util.Collections;
+
+@Slf4j
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping(value = "/api/file")
+public class FileController {
+
+    private final FileService fileService;
+
+    public FileController(FileService fileService) {
+        this.fileService = fileService;
+    }
+
+    @PostMapping
+    public ResponseEntity getCustomersOnFile(@RequestHeader(name = "x-school-org-id") String schoolId, @RequestBody byte[] file) {
+        CustomerFileGroup customersFromFile;
+
+        String contentType = new Tika().detect(file);
+
+        if (fileService.isTypeOfTypeExcel(contentType)) {
```

Comment:
Also the detection of valid file type should be delegated to `FileService` and be thrown as an Exception.

Reply:
Fixed